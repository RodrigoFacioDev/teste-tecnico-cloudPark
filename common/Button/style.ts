import styled, { css } from "styled-components/native";

export const StyledButton = styled.TouchableOpacity<{
  variant?: "container" | "ghost";
  color?: string;
}>`
  width: 100%;
  padding: 12px;
  border-radius: 8px;
  align-items: center;
  ${({ variant, color }) =>
    variant === "ghost"
      ? css`
          background-color: transparent;
          border-width: 0;
        `
      : css`
          background-color: ${color || "#398ccb"};
        `}
`;

export const ButtonText = styled.Text<{
  variant?: "container" | "ghost";
  color?: string;
}>`
  font-size: 18px;
  font-weight: bold;
  ${({ variant, color }) =>
    variant === "ghost"
      ? css`
          color: ${color || "#398ccb"};
        `
      : css`
          color: #fff;
        `}
`;
