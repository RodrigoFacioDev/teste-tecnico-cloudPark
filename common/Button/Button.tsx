import React from "react";
import { TouchableOpacityProps } from "react-native";
import { ButtonText, StyledButton } from "./style";

interface ButtonProps extends TouchableOpacityProps {
  children: React.ReactNode;
  variant?: "container" | "ghost";
  color?: string;
}

export default function Button({
  children,
  variant = "container",
  color,
  ...rest
}: ButtonProps) {
  return (
    <StyledButton variant={variant} color={color} {...rest}>
      <ButtonText variant={variant} color={color}>
        {children}
      </ButtonText>
    </StyledButton>
  );
}
