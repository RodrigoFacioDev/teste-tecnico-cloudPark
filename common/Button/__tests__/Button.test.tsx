import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import Button from "../Button";

describe("<Button />", () => {
  it("should render the text when children is provided", () => {
    const { getByText } = render(<Button>Save</Button>);
    expect(getByText("Save")).toBeTruthy();
  });

  it("should call onPress when pressed", () => {
    const onPressMock = jest.fn();
    const { getByText } = render(<Button onPress={onPressMock}>Click</Button>);
    fireEvent.press(getByText("Click"));
    expect(onPressMock).toHaveBeenCalled();
  });

  it("should match the component snapshot when rendered", () => {
    const tree = render(<Button>Snapshot</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
