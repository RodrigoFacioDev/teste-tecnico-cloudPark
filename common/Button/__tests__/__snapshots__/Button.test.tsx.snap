// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> faz snapshot do componente 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#398ccb",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
    variant="container"
  >
    Snapshot
  </Text>
</View>
`;

exports[`<Button /> matches the component snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#398ccb",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
    variant="container"
  >
    Snapshot
  </Text>
</View>
`;

exports[`<Button /> should match the component snapshot when rendered 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#398ccb",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
    variant="container"
  >
    Snapshot
  </Text>
</View>
`;
